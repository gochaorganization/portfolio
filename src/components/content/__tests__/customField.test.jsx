import React from 'react'
import { shallow, mount } from 'enzyme'
import CustomField from '../customField'

describe('CustomField tests', () => {
  const defaultProps = {
    label: 'Label',
    meta: { touched: false, error: '' },
  }

  const errorProps = {
    meta: { touched: true, error: 'Error occured' },
  }

  it('should render correctly', () => {
    const content = shallow(<CustomField {...defaultProps} />)
    expect(content).toMatchSnapshot()
  })

  it('should render title', () => {
    const content = shallow(<CustomField {...defaultProps} />)
    expect(content.find('.form-field__description').text()).toEqual(
      defaultProps.label
    )
  })

  it('should show error', () => {
    const content = shallow(<CustomField {...errorProps} />)
    expect(content.find('.form-field__error').text()).toEqual(
      errorProps.meta.error
    )
  })
})
