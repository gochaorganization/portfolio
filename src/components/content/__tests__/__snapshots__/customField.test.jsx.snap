// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomField tests should render correctly 1`] = `ShallowWrapper {}`;
